### PR TITLE
Feat/test parser

### PR DESCRIPTION
**Pull request by scottnath**
#### Changelog

**New**

- adds test-parser
- adds fixtures for test-parser tests
- adds `chai` as assertion library

### CODE REVIEW

File: src/__tests__/fixtures/div-requirements.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  selectors: {
+    root: 'div',
+    content: 'span',
+  },
+};
+
+/**
+ * Grabs Div component elements from a DOM
+ * @param {DocumentFragment} docFragment - JavaScript document or JSDOM fragment
+ * @param {buttonElements} selectors
```
Comment (elizabethsjudd) [inline src/__tests__/fixtures/div-requirements.js]: C&P typo

File: src/__tests__/fixtures/div-requirements.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      defaults.content.text
+    );
+  },
+};
+
+/**
+ * Returns a set of test objects
+ * @return {scenario-test-object[]} test objects for use in `testRunner`
+ */
+export const FixtureTests = () => {
```
Comment (elizabethsjudd) [inline src/__tests__/fixtures/div-requirements.js]: I know this is a "fixture" file but aren't they suppose to mimic actual setups... if that's the case why would this function be called `FixtureTests` . Wouldn't it be `DivTests`?

File: src/test-parser.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
+/* global it, describe, before, beforeEach */
+
+/**
+ * Scenario test object structure
+ * @typedef {object} scenario-test-object
+ * @property {string} scenario - Scenario name
+ * @property {function} [given] - Given/When statement that determines if test should be ran
```
Comment (elizabethsjudd) [inline src/test-parser.js]: remove `/When` from description

**Review by elizabethsjudd — COMMENTED** [review]
Three very small documentation changes.  Otherwise it looks good!

**Comment by scottnath** [comment]
@elizabethsjudd PR changes implemented

**Comment by elizabethsjudd** [comment]
@scottnath the changes look good.  I wonder if this should be merging with `master` though. I feel like we should keep our forked `masters` consistent with carbon. I created a branch `feat/test-parser` in my fork if you want to switch the PR to.

**Comment by scottnath** [comment]
@elizabethsjudd base changed

**Review by elizabethsjudd — APPROVED** [review]

**Pull request merged**